x-pack/filebeat/input/salesforce/input.go

```diff
@@ -0,0 +1,506 @@
+		if s.cursor.Object.FirstEventTime != "" {
+			object.Put("first_event_time", s.cursor.Object.FirstEventTime)
+		}
+		if s.cursor.Object.LastEventTime != "" {
+			object.Put("last_event_time", s.cursor.Object.LastEventTime)
+		}
+		cursor = mapstr.M{"object": object}
+	}
+	query, err := s.FormQueryWithCursor(s.config.EventMonitoringMethod.Object.Query, cursor)
+	if err != nil {
+		return fmt.Errorf("error forming based on cursor: %w", err)
+	}
+
+	res, err := s.soqlr.Query(query, false)
+	if err != nil {
+		return err
+	}
+
+	totalEvents := 0
+	firstEvent := true
+
+	for res.TotalSize() > 0 {
+		for _, rec := range res.Records() {
+			val := rec.Record().Fields()
+
+			jsonStrEvent, err := json.Marshal(val)
+			if err != nil {
+				return err
+			}
+
+			if firstEvent {
+				if timstamp, ok := val[s.config.EventMonitoringMethod.Object.Cursor.Field].(string); ok {
+					s.cursor.Object.FirstEventTime = timstamp
+				}
+			}
+
+			if timstamp, ok := val[s.config.EventMonitoringMethod.Object.Cursor.Field].(string); ok {
+				s.cursor.Object.LastEventTime = timstamp
+			}
+
+			err = publishEvent(s.publisher, s.cursor, jsonStrEvent, "Object")
+			if err != nil {
+				return err
+			}
+			firstEvent = false
+			totalEvents++
+		}
+
+		if res.MoreRecords() { // returns true if there are more records.
+			res, err = res.Next()
+			if err != nil {
+				return err
+			}
+		} else {
+			break
+		}
+	}
+	s.log.Debugf("Total events: %d", totalEvents)
+
+	return nil
+}
+
+// RunEventLogFile runs the EventLogFile method of the Event Monitoring API to
+// collect events.
+func (s *salesforceInput) RunEventLogFile() error {
+	s.log.Debugf("Scrape EventLogFiles every %s", s.srcConfig.EventMonitoringMethod.EventLogFile.Interval)
+
+	var cursor mapstr.M
+	if !(s.cursor.EventLogFile.FirstEventTime == "" && s.cursor.EventLogFile.LastEventTime == "") {
+		eventLogFile := make(mapstr.M)
+		if s.cursor.EventLogFile.FirstEventTime != "" {
+			eventLogFile.Put("first_event_time", s.cursor.EventLogFile.FirstEventTime)
+		}
+		if s.cursor.EventLogFile.LastEventTime != "" {
+			eventLogFile.Put("last_event_time", s.cursor.EventLogFile.LastEventTime)
+		}
+		cursor = mapstr.M{"event_log_file": eventLogFile}
+	}
+
+	query, err := s.FormQueryWithCursor(s.config.EventMonitoringMethod.EventLogFile.Query, cursor)
+	if err != nil {
+		return fmt.Errorf("error forming based on cursor: %w", err)
+	}
+
+	res, err := s.soqlr.Query(query, false)
+	if err != nil {
+		return err
+	}
+
+	if s.sfdcConfig.Client == nil {
+		return errors.New("internal error: salesforce configuration is not set properly")
+	}
+
+	totalEvents, firstEvent := 0, true
+	for res.TotalSize() > 0 {
+		for _, rec := range res.Records() {
+			req, err := http.NewRequestWithContext(s.ctx, http.MethodGet, s.config.URL+rec.Record().Fields()["LogFile"].(string), nil)
+			if err != nil {
+				return err
+			}
+
+			s.clientSession.AuthorizationHeader(req)
+
+			req.Header.Add("X-PrettyPrint", "1")
```

Comment:
is this necessary?

Reply:
Good point. This came up when Kush and I were also reviewing the code. Seems very safe to remove as this just gives pretty JSON?

But see this: https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/dome_event_log_file_download.htm?q=X-PrettyPrint%3A1

So, we decided to keep it. But yes, I'll add a comment there with this link.

Reply:
Yes, it works. Only kept it because of what we saw here: https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/dome_event_log_file_download.htm?q=X-PrettyPrint%3A1

I removed it, and we'll reintroduce the header if needed (if someone files a related bug report).